### PR TITLE
add function to run amcheck and check for db corruption

### PR DESCRIPTION
**Pull request by KPostOffice**
## Related Issues and Dependencies

https://github.com/thoth-station/storages/issues/1885

## This introduces a breaking change

- [ ] Yes
- [x] No

## This should yield a new module release

- [x] Yes
- [ ] No

## This Pull Request implements

Runs amcheck to check for corruption

## Description

1. Check if amcheck is installed
    1. if amcheck is not installed, install it
1. run amcheck, if there is an error return true otherwise return false

### CODE REVIEW

File: thoth/storages/graph/postgres.py [hunk 1]
```diff
@@ -5363,6 +5363,43 @@ def cache_clear(self) -> None:
...
+
+        with self._session_scope() as session:
+            if session.execute(check_for_ext_query) == []:
+                session.execute(install_ext_query)
+
+            try:
+                session.execute(run_amcheck_query)
+                return False
+            except:
+                return True
```
Comment (fridex) [inline thoth/storages/graph/postgres.py:5401]: Just a minor comment. Can we log exception here? Might be useful for other failure scenarios.

**Review by fridex — COMMENTED** [review]
:1st_place_medal: :1st_place_medal: :1st_place_medal: one minor comment, lgtm otherwise

File: thoth/storages/graph/postgres.py (hunk 1 above)
Comment (KPostOffice) [reply thoth/storages/graph/postgres.py:5401]: Yeah for sure ++

**Review by fridex — APPROVED** [review]
:100: :+1:

**Pull request merged**